CMakeLists.txt

```diff
@@ -52,56 +59,196 @@ if (NOT HAVE_CXX_ATOMICS_WITHOUT_LIB
+    if(NOT MSVC)
+        add_definitions(-D_WIN32_WINNT=${WINNT_VERSION})
+    endif()
+
+    set(CMAKE_DEBUG_POSTFIX "")
+endif()
+
 list (APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 if (NOT MSVC)
-    set (THREADS_PREFER_PTHREAD_FLAG TRUE)
-    find_package (Threads)
-    find_package (PkgConfig REQUIRED)
-    pkg_search_module (GnuTLS REQUIRED IMPORTED_TARGET gnutls)
-    pkg_search_module (Nettle REQUIRED IMPORTED_TARGET nettle)
-    check_include_file_cxx(msgpack.hpp HAVE_MSGPACKCXX)
-    if (NOT HAVE_MSGPACKCXX)
-        find_package(msgpack QUIET CONFIG NAMES msgpack msgpackc-cxx)
-        if (NOT msgpack_FOUND)
-            find_package(msgpack REQUIRED CONFIG NAMES msgpack-cxx)
-            set(MSGPACK_TARGET "msgpack-cxx")
-        else()
-            set(MSGPACK_TARGET "msgpackc-cxx")
+    if (WIN32 AND MINGW) # MSYS2
+        set (THREADS_PREFER_PTHREAD_FLAG TRUE)
+        find_package (Threads)
+        set(CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO RELEASE)
+        if(DEFINED $ENV{BOOST_ROOT})
+            set(BOOST_ROOT "$ENV{BOOST_ROOT}")
         endif()
-    endif()
-    if (OPENDHT_TOOLS)
-        find_package (Readline 6 REQUIRED)
-    endif ()
-    pkg_search_module(argon2 REQUIRED IMPORTED_TARGET libargon2)
-    set(argon2_lib ", libargon2")
-    pkg_search_module(Jsoncpp IMPORTED_TARGET jsoncpp)
-    if (Jsoncpp_FOUND)
+        if(DEFINED BOOST_ROOT)
+            set(Boost_DIR "${BOOST_ROOT}")
+        endif()
+        find_package(Boost REQUIRED)
+        include_directories(${Boost_INCLUDE_DIRS})
+        link_directories(${Boost_LIBRARY_DIRS})
+        
+        find_path(ARGON2_INCLUDEDIR argon2.h REQUIRED)
+        include_directories(${ARGON2_INCLUDEDIR})
+        
+        set(FETCHCONTENT_QUIET FALSE)
+        if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+            FetchContent_Declare(gnutls
+                URL "https://www.gnupg.org/ftp/gcrypt/gnutls/v3.8/gnutls-3.8.4-w64.zip"
+            )
+            set (GNUTLS_X64 1)
+        elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
+            FetchContent_Declare(gnutls
+                URL "https://www.gnupg.org/ftp/gcrypt/gnutls/v3.8/gnutls-3.8.9-w32.zip"
+            )
+            unset (GNUTLS_X64)
+        endif()
+        FetchContent_Declare(cJSON
+            GIT_REPOSITORY "https://github.com/DaveGamble/cJSON.git"
+            OVERRIDE_FIND_PACKAGE
+        ) # Dependency for msgpack
+        FetchContent_Declare(msgpack-c
+            GIT_REPOSITORY "https://github.com/msgpack/msgpack-c.git"
+            GIT_TAG        "cpp-6.1.1"
+            FIND_PACKAGE_ARGS NAMES msgpack-c
+        )
+        FetchContent_Declare(jsoncpp
+            GIT_REPOSITORY "https://github.com/open-source-parsers/jsoncpp.git"
+            GIT_TAG        "1.9.5"
+            OVERRIDE_FIND_PACKAGE
+        )
+        FetchContent_Declare(fmt
+            GIT_REPOSITORY "https://github.com/fmtlib/fmt.git"
+            GIT_TAG        "11.1.2"
+            OVERRIDE_FIND_PACKAGE
```

Comment:
avoid FetchContent, this PR may simply make the build work well in a vcpkg environment.